src/Widgets/TerminalWidget.vala

```diff
@@ -129,10 +130,17 @@ namespace PantheonTerminal {
             /* Connect to necessary signals */
             button_press_event.connect ((event) => {
                 if (event.button ==  Gdk.BUTTON_SECONDARY) {
+                    // Reset the state of buttons when we lunch the context menu again, and we have a uri
+                    if (uri != null) {
+                        window.get_simple_action (PantheonTerminalWindow.ACTION_COPY).set_enabled (false);
+                        this.menu.get_children ().nth_data (0).set_visible (false);
+                    }
```

Comment:
Why do you need this? You reset uri and the action in the following code.

Reply:
Yeah so this is an old not current behaviour, if you make the context menu on top of uri you will got the copy enabled, but if you close it and open it somewhere else no one changes the copy state so it will stay enabled, but It should not be enabled as there's nothing selected and we are not on top of uri

Reply:
Sorry yes, you are right. The first clause could be true and the second false (or vice versa) so both are needed.